### ISSUE TITLE
Create a JSBI → BigInt transpiler

### ISSUE THREAD

**Issue by mathiasbynens** [GoogleChromeLabs/jsbi#2]
Most features are transpiled from their standardized form into a more compatible form. For BigInt specifically, this is not feasible for [the reasons outlined in the README](https://github.com/GoogleChromeLabs/jsbi#readme).

As such, developers wishing to use BigInt functionality today should instead use JSBI to write their code. Then, when the time comes that BigInts are natively supported everywhere, they should be able to transpile their JSBI-based code into native BigInt code.

In other words, we need a JSBI → native BigInt transpiler.

If you’d like to help, please speak up!

**Update:** @FWeinb created [a prototype Babel plugin](https://github.com/GoogleChromeLabs/jsbi/issues/2#issuecomment-437473106). What’s left to do is for someone to take that code, package it up as its own repository with tests and publish it as an npm module.

**Comment by mathiasbynens** [comment]
For example, the README demo:

```js
const max = JSBigInt(Number.MAX_SAFE_INTEGER);
console.log(String(max));
// → '9007199254740991'
const other = JSBigInt('2');
const result = max.add(other);
console.log(String(result));
// → '9007199254740993'
```

…would be transpiled to…

```js
const max = BigInt(Number.MAX_SAFE_INTEGER);
console.log(String(max));
// → '9007199254740991'
const other = BigInt('2');
const result = max + other;
console.log(String(result));
// → '9007199254740993'
```

**Comment by FWeinb** [comment]
Transpiling this will get very tricky. See this example:
```js
const addNumbers = (a, b) => a.add(b) // 
const a = JSBigInt('1');
const b = JSBigInt('2')
const result = addNumbers(a, b);
``
It would be very hard to statically analysis the code and track all functions that take JSBigInts.

**Comment by mathiasbynens** [comment]
That is indeed the main challenge. You can’t just replace all calls to a method named `add`, for example. I want to believe some tooling exists already to do this kind of tracking, though.

**Comment by FWeinb** [comment]
Here is a quick prototype with using the static api https://astexplorer.net/#/gist/f3b0272a37d21487e8b3561c1c7d38c2/337ce3b1b66672c543cd77c85f912fa6202f3d60

**Comment by mathiasbynens** [comment]
I chatted with @hzoo today who confirmed that although there are a few existing projects in this area, it would certainly be easier to just change JSBI’s API to turn everything into static methods. So let’s just do that. @FWeinb’s WIP prototype could then be turned into a proper Babel plugin :)

**Comment by mathiasbynens** [comment]
The API was changed in #10, and jsbi@2.0.0 includes the changes. @FWeinb, would you be interested in packaging up your transpiler into a fully-fledged GitHub repo of its own + publish it on npm?

**Comment by FWeinb** [comment]
@mathiasbynens I currently don't have enough time to maintain a babel plugin. Feel free to use the code I posted as a baseline.

**Comment by mathiasbynens** [comment]
I’ve updated @FWeinb’s prototype per the latest API changes: https://astexplorer.net/#/gist/f3b0272a37d21487e8b3561c1c7d38c2/797f79cb0e7c1f8251d6e450c65e3c3cea73bb0c

**Comment by mathiasbynens** [comment]
@FWeinb Would you mind signing [the CLA](https://cla.developers.google.com/clas) so we can release the plugin based on your prototype under the same license as JSBI itself?

**Comment by mathiasbynens** [comment]
More updates: https://astexplorer.net/#/gist/f3b0272a37d21487e8b3561c1c7d38c2/d2f41b52a55a06c4a00951f8953f0fb58317ab3d

**Comment by FWeinb** [comment]
@mathiasbynens I signed the CLA.

**Comment by mathiasbynens** [comment]
We’re working on publishing this soon.

**Comment by mathiasbynens** [comment]
https://github.com/GoogleChromeLabs/babel-plugin-transform-jsbi-to-bigint  🔥

Thanks for the help everyone! Special shout-out to @FWeinb for putting together the prototype and allowing us to build upon it.
